### PR TITLE
Vagrant installation (#64)

### PR DESCRIPTION
**Pull request by rugabarbo**
Please, review and try it yourself.
See #64.

### CODE REVIEW

File: vagrant/provision/once.sh [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+debconf-set-selections <<< "mysql-server-5.6 mysql-server/root_password password \"''\""
+debconf-set-selections <<< "mysql-server-5.6 mysql-server/root_password_again password \"''\""
+echo "Done!"
+
+info "Update OS software"
+sudo apt-get update
+sudo apt-get upgrade -y
+
+info "Install additional software"
+sudo apt-get install -y git php5-cli php5-intl php5-mysqlnd php5-gd php5-fpm nginx mysql-server-5.6
```
Comment (SilverFire) [inline vagrant/provision/once.sh]: xdebug might me added

File: docs/guide/start-installation.md [hunk 2]
```diff
@@ -5,6 +5,30 @@ Installation
 
 The minimum requirement by this project template is that your Web server supports PHP 5.4.0.
 
+## Installing using Vagrant
+
+This way is easiest but long (~20 min).
```
Comment (SilverFire) [inline docs/guide/start-installation.md]: `is easiest` -> `is the easiest`

File: docs/guide/start-installation.md [hunk 3]
```diff
@@ -5,6 +5,30 @@ Installation
...
+
+1. Install [VirtualBox](https://www.virtualbox.org/wiki/Downloads)
+2. Install [Vagrant](https://www.vagrantup.com/downloads.html)
+3. Create GitHub [personal API token](https://github.com/blog/1509-personal-api-tokens)
+4. Run commands:
+
+   ```
```
Comment (SilverFire) [inline docs/guide/start-installation.md]: \`\`\`bash

File: docs/guide/start-installation.md [hunk 4]
```diff
@@ -5,6 +5,30 @@ Installation
...
+   vagrant plugin install vagrant-hostmanager
+   ```
+5. Place your GitHub personal API token to `vagrant/config/vagrant-local.yml`
+6. Run command:
+
+   ```
+   vagrant up
+   ```
+
+That`s all. You just need to wait for completion.
```
Comment (SilverFire) [inline docs/guide/start-installation.md]: That\`s -> `That's`

Using backtick causes incorrec patsing

File: vagrant/nginx/app.conf [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+   # uncomment to avoid processing of calls to non-existing static files by Yii
+   #location ~ \.(js|css|png|jpg|gif|swf|ico|pdf|mov|fla|zip|rar)$ {
+   #    try_files $uri =404;
+   #}
+   #error_page 404 /404.html;
+
+   location ~ \.php$ {
+       include fastcgi_params;
+       fastcgi_param SCRIPT_FILENAME $document_root/$fastcgi_script_name;
```
Comment (SilverFire) [inline vagrant/nginx/app.conf]: might be useful to add

```nginx
    fastcgi_param   PHP_VALUE "error_log=/var/log/php/$host.log";
```

to have the separated logs for backend and frontend

**Comment by SilverFire** [comment]
Thank you very much for your effort!
Unfortunately, I can't check it with Vagrant now, so I would like to ask someone who's reading this to try, if you are interested.

**Comment by rugabarbo** [comment]
Thank you for review! It'll be great if you'll ask someone to test. I'm bad QA-enginier :) I checked it under Mac OS X (10.10.5) only.

File: vagrant/nginx/app.conf (hunk 5 above)
Comment (rugabarbo) [inline vagrant/nginx/app.conf]: I use standard Yii config for nginx. I don't think we must customize it here. Each user can customize it (after git clone) for own requirements...

File: vagrant/provision/once.sh (hunk 1 above)
Comment (rugabarbo) [inline vagrant/provision/once.sh]: I think we should test current realization. And if it OK, merge it. After that it'll be comfortable to add new functionality (we'll open other issues/PRs for them).

**Comment by rugabarbo** [comment]
We must write additional docs for Windows users, because they have no bash.

**Comment by rugabarbo** [comment]
Docs for Windows users are ready. I'll test them on non-virutal Windows machine tomorrow.

**Comment by rugabarbo** [comment]
If smbd want test it, you can download enhanced distrib (step 5) from [here](https://github.com/rugabarbo/yii2-app-advanced/archive/vagrant.zip)

**Comment by rugabarbo** [comment]
@SilverFire, I checked it under Win7x64 and changed instructions. It works! 
Dmitry, please, help me correct English grammar for installation readme and then we can merge it.

File: docs/guide/start-installation.md [hunk 6]
```diff
@@ -5,6 +5,53 @@ Installation
...
+      cp vagrant-local.example.yml vagrant-local.yml
+      ```
+   2. Windows users:
+      * download repo [yii2-app-advanced](https://github.com/yiisoft/yii2-app-advanced/archive/master.zip)
+      * unzip it
+      * go into directory `yii2-app-advanced-master/vagrant/config`
+      * copy `vagrant-local.example.yml` to `vagrant-local.yml`
+   
+6. Place your GitHub personal API token to `vagrant-local.yml`
+7. Windows users must manually add next strings to [hosts file](https://ru.wikipedia.org/wiki/Hosts):
```
Comment (SilverFire) [inline docs/guide/start-installation.md]: `add next strings` -> `add the following lines`

**Comment by rugabarbo** [comment]
@SilverFire, thanks!
Fixed.

I'm ready for merge, so I'm waiting to merge or your feedback about this PR.

**Comment by rugabarbo** [comment]
@SilverFire, I am preparing improved version for this PR. Don't merge it until I update it, please.

**Comment by SilverFire** [comment]
OK

**Comment by rugabarbo** [comment]
It's ready.

**Comment by SilverFire** [comment]
@yiisoft/core-developers I'd like to include Vagrant installation in the release, but I don't want to add Vagrant files to this repo. I suggest to create a separate repo `vagrant-yii2-app-advanced`, put there code, configs and docs.

File: docs/guide/start-installation.md [hunk 7]
```diff
@@ -5,6 +5,53 @@ Installation
...
+      cp vagrant-local.example.yml vagrant-local.yml
+      ```
+   2. Windows users:
+      * download repo [yii2-app-advanced](https://github.com/yiisoft/yii2-app-advanced/archive/master.zip)
+      * unzip it
+      * go into directory `yii2-app-advanced-master/vagrant/config`
+      * copy `vagrant-local.example.yml` to `vagrant-local.yml`
+   
+6. Place your GitHub personal API token to `vagrant-local.yml`
+7. Windows users must manually add the following lines to [hosts file](https://ru.wikipedia.org/wiki/Hosts):
```
Comment (agungsijawir) [inline docs/guide/start-installation.md]: Should use `https://en.wikipedia.org/wiki/Host` with English instead `https://ru.wikipedia.org/wiki/Hosts` ?

Comment (rugabarbo) [inline docs/guide/start-installation.md]: Fixed. Thanks!

**Comment by SilverFire** [comment]
I'm about to merge this PR. @yiisoft/core-developers any comments?

**Comment by samdark** [comment]
Not from me. I'm not using Vagrant.

**Comment by cebe** [comment]
fine with me.

**Comment by SilverFire** [comment]
@rugabarbo Thank you very much!

**Comment by rugabarbo** [comment]
@SilverFire you are welcome :)

**Pull request merged**